### DISCUSSION TITLE
Default stride for pooling functions

### DISCUSSION THREAD

**Discussion by dogeplusplus** [google/flax#2023] [Ideas] [answered]
A bit of a nitpick but I was wondering why the default behavior of pooling functions is to have stride 1 instead of the `window_shape`? I feel that for most use cases the stride would be the dimension of the kernel size as in other frameworks.

**Comment by marcvanzee** [comment, marked as answer]
Thanks for bringing this up! 

**Update 25 Apr, 2022:** #2064 adds default stride to the docstring of pooling.
~~We should document these defaults in our docstring, because right now we don't, which is very odd actually: https://github.com/google/flax/blob/main/flax/linen/pooling.py#L36~~

I took a brief look at other frameworks for `max_pool`:
* For Pytorch indeed the default seems to be kernel size: https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html
* For TF it seems `strides` is not an optional argument, and doesn't have a default value: https://www.tensorflow.org/api_docs/python/tf/nn/max_pool

I actually don't know why we chose this default, it does seem a bit surprising if you come from torch. (@jheek any idea?).

**Comment by AnirudhDagar** [comment]
I was wondering the same about this default choice. Would be nice to know if there is a good reason behind it.
